consensus/src/experimental/buffer_manager.rs

```diff
@@ -202,12 +206,14 @@ impl StateManager {
                     .map(Arc::new)
                     .collect();
                 // send to persisting phase
-                block_on(self.persisting_phase_tx.send(PersistingRequest {
-                    blocks,
-                    commit_ledger_info: aggregated.aggregated_proof,
-                    callback: aggregated.callback,
-                }))
-                .ok();
+                self.persisting_phase_tx
+                    .send(PersistingRequest {
+                        blocks,
+                        commit_ledger_info: aggregated.aggregated_proof,
+                        callback: aggregated.callback,
+                    })
+                    .await
```

Comment:
This future await is dangerous that we need make sure no other async task will visit this item before it yields. Right now, this function is directly called from the main loop, so it is okay.

Reply:
we should move all channels to sync in following pr